### ISSUE TITLE
The IntervalIndex documentation makes no mention of IntervalIndex.get_loc

### ISSUE THREAD

**Issue by dumbledad** [pandas-dev/pandas#31073]
#### Code Sample, a copy-pastable example if possible

```python
# No code, documentation issue
```
#### Problem description

Why doesn't the documentation about IntervalIndex ([here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.IntervalIndex.html)) contain any mention of the function [IntervalIndex.get_loc](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.IntervalIndex.get_loc.html)?

#### Expected Output

I would expect the list of class methods to include all the methods.

**Comment by MarcoGorelli** [comment]
Thanks for the report - it's documented in [Index](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.html#pandas.Index), from which IntervalIndex inherits. In `see also` we mention:
```
Index
    The base pandas Index type.
```

**Comment by charlesdong1991** [comment]
https://pandas.io/docs/reference/indexing.html#intervalindex

might be able to see it on new pandas website

**Comment by jorisvandenbossche** [comment]
Yes, the IntervalIndex class docstring page (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.IntervalIndex.html) doesn't include this on purpose, it only lists the methods and attributes which are specific to IntervalIndex.

**Comment by jschendel** [comment]
I could see an argument for including these methods for `IntervalIndex` since the behavior is non-standard.  Other index types require exact matches for indexing methods like `get_loc`/`get_indexer` but `IntervalIndex` has the additional flexibility for matching on points contained in the intervals:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '1.0.0rc0+134.gbbcda98c7'

In [2]: ii = pd.interval_range(0, 4)

In [3]: ii
Out[3]: 
IntervalIndex([(0, 1], (1, 2], (2, 3], (3, 4]],
              closed='right',
              dtype='interval[int64]')

In [4]: ii.get_loc(ii[3])
Out[4]: 3

In [5]: ii.get_loc(3.5)
Out[5]: 3
```
So in the above `Out[4]` is consistent with the standard behavior associated with other index types but `Out[5]` is not.

Not sure if this is a good enough reason to explicitly include these in the `IntervalIndex` class docstring page but it does illustrate that there is some `IntervalIndex` specific behavior with to these methods.

**Comment by jorisvandenbossche** [comment]
@jschendel that sounds as a good reason to include those index methods in the API yes
